src/pow.cpp

```diff
@@ -15,30 +15,21 @@
 
 unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHeader *pblock, const Consensus::Params& params)
 {
+    // MVF-BU begin difficulty re-targeting
+    if (params.MVFisWithinRetargetPeriod(pindexLast->nHeight+1))
+        return GetMVFNextWorkRequired(pindexLast, pblock, params);
+    // MVF-BU end
+
     unsigned int nProofOfWorkLimit = UintToArith256(params.powLimit).GetCompact();
 
     // Genesis block
     if (pindexLast == NULL)
         return nProofOfWorkLimit;
 
-    // MVF-BU begin difficulty re-targeting reset (MVHF-BU-DES-DIAD-2)
-    if (pindexLast->nHeight == FinalActivateForkHeight)
-    {
-        arith_uint256 bnNew;
-
-        bnNew.SetCompact(0x207eeeee);
-        //bnNew *= 10;     // drop difficulty by factor of 10
-
-        LogPrintf("MVF FORK BLOCK DIFFICULTY RESET  %08x  \n", bnNew.GetCompact());
-        return bnNew.GetCompact();
-    }
-    LogPrintf("MVF DEBUG DifficultyAdjInterval = %d , TargetTimeSpan = %d \n", params.DifficultyAdjustmentInterval(pindexLast->nHeight), params.MVFPowTargetTimespan(pindexLast->nHeight));
-    // MVF-BU end
-
     // Only change once per difficulty adjustment interval
-    if ((pindexLast->nHeight+1) % params.DifficultyAdjustmentInterval(pindexLast->nHeight) != 0)  // MVF-BU: use height-dependent interval
+    if ((pindexLast->nHeight+1) % params.DifficultyAdjustmentInterval(pindexLast->nHeight) != 0)
```

Comment:
This doesn't look like it's been reverted completely to pre-MVF state.
Since it's only called if not in the fork retarget range, is it possible to restore it completely (maybe use  a default argument for the DifficultyAdjustmentInterval function so that if we don't pass a height, it behaves same as before) ?